planar/triangulate/triangulate.go

```diff
@@ -0,0 +1,16 @@
+package triangulate
+
+import "github.com/go-spatial/geom"
+
+type Interface interface {
+	// SetPoints sets the nodes to be used in the triangulation
+	SetPoints(pts []geom.Point, data []interface{})
+	// Triangles returns the triangles that are produced by the triangulation
+	Triangles() []geom.Triangle
+}
+
+type Constrainer interface {
```

Comment:
This should probably come with a bit more description. Maybe reference Constrained Delaunay triangulation?